Documentation/operation docs rework (#3074)
scality/metalk8s: docs/operation/account_administration/user_authorization.rst

User Authorization
==================

Kubernetes API
--------------

> To authorize specific users and/or groups against Kubernetes API, the

Reviewer: ```suggestion
To authorize users and groups against the Kubernetes API, the
```

:term:`API Server` relies on *RBAC* (Role-Based Access Control), through the
use of special API objects:

> - **(Cluster)Roles**, which define specific permissions on a set of API

Reviewer: ```suggestion
- **Roles** and **ClusterRoles**, which define specific permissions on a set of API
```

  resources,

> - **(Cluster)RoleBindings**, which map a user or group to a set of

Reviewer: ```suggestion
- **RoleBindings** and **ClusterRoleBindings**, which map a user or group to a set of
```

>   (Cluster)Roles.

Reviewer: Roles or ClusterRoles.

.. note::

   MetalK8s includes pre-provisioned ClusterRoles. Platform administrators can

>    create new (Cluster)Roles or refer to existing ones.

Reviewer: ```suggestion
   create new Roles or ClusterRoles or refer to existing ones.
```

ClusterRoles
************

- Obtain the list of available ClusterRoles.

  .. code-block:: shell

     root@bootstrap $ kubectl --kubeconfig /etc/kubernetes/admin.conf \
                          get clusterroles

- Describe a ClusterRole for more information.

  .. code-block:: shell

     root@bootstrap $ kubectl --kubeconfig /etc/kubernetes/admin.conf \
                          describe clusterrole <name>

- The pre-provisioned static user **admin@metalk8s.invalid** is already bound
  to the **cluster-admin** ClusterRole, which grants cluster-wide permissions
  to all exposed APIs.

  .. code-block:: shell

     root@bootstrap $ kubectl --kubeconfig /etc/kubernetes/admin.conf \
                          describe clusterrole cluster-admin
     Name:         cluster-admin
     Labels:       kubernetes.io/bootstrapping=rbac-defaults
     Annotations:  rbac.authorization.kubernetes.io/autoupdate: true
     PolicyRule:
       Resources  Non-Resource URLs  Resource Names  Verbs
       ---------  -----------------  --------------  -----
       *.*        []                 []              [*]
                  [*]                []              [*]

> For more information on Kubernetes authorization mechanisms, refer to the

Reviewer 2: Kubernetes'

Reviewer: I disagree. This makes sense if not possessive. If possessive, the default would be "Kubernetes's" (though I could see a spirited discussion arising on this point...)

`RBAC <https://kubernetes.io/docs/reference/access-authn-authz/rbac/>`_
documentation.

.. _bind-user-to-role:

> (Cluster)RoleBindings

Reviewer: ```suggestion
ClusterRoleBindings
```
